### PR TITLE
Added access control for direct communication

### PR DESCRIPTION
**Pull request by favreau**

### CODE REVIEW

**Comment by eile** [comment]
Why did you close #131?

File: doc/feature/restAPI.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    * Response:
+```
+    {
+      { "namespace/foo": ["GET", "PUT"] },
+      { "namespace/bar": ["PUT"] }
+    }
+```
+  - Get JSON schema for a specific object:
+    * Request: ```GET http://hostname:port/namespace/foo/schema```
+    * Response:```{ "foo": { "name": string, "age": integer } }```
```
Comment (eile) [inline doc/feature/restAPI.md]: foo -> Foo (class names are capitalized

Response is foo : schema or just schema, i.e., ``` { "name": string, "age": integer }```?

File: doc/feature/restAPI.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    _httpServer = zeq::http::Server::parse( argc, argv );
+    if( !_httpServer )
+      return;
+
+    // Register C++ objects
+    subscribers.push_back( _httpServer );
+    _httpServer.subscribe( camera, LUT, vrParams );
+    _httpServer.register( camera, LUT, vrParams, frame );
+  }
+```
```
Comment (eile) [inline doc/feature/restAPI.md]: Part of the http server spec - Example here should show a schema query.

File: doc/feature/restAPI.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+Query API:
+   * Request: ```GET http://hostname:port/zerobuf/render/camera/schema```
+   * Response: ```{ "camera" : { "origin": "string", "lookat": "up" } }```
```
Comment (eile) [inline doc/feature/restAPI.md]: Repeated?

File: doc/feature/restAPI.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+## Implementation
+- Add a getSchema public method to Servus::serializable
+- Make sure getSchema is implemented by to the code generator and returns the
```
Comment (eile) [inline doc/feature/restAPI.md]: rm to

**Comment by favreau** [comment]
Because I am a *&^ github *@#$

File: zeroeq/http/server.cpp [hunk 5]
```diff
@@ -25,6 +25,19 @@ std::string _toLower( std::string value )
     std::transform( value.begin(), value.end(), value.begin(), ::tolower );
     return value;
 }
+
+/**
+  In a typical situation, user agents can discover via a preflight request
+  whether a cross-origin resource is prepared to accept requests. The current
+  implementation of this class does not support this mechanism and has to
+  accept cross-origin resources. In order to acheive that, access control
```
Comment (eile) [inline zeroeq/http/server.cpp]: typo achieve

File: zeroeq/http/server.cpp [hunk 6]
```diff
@@ -25,6 +25,19 @@ std::string _toLower( std::string value )
...
+  headers are added to all HTTP responses.
```
Comment (eile) [inline zeroeq/http/server.cpp]: + Right now we accept all sources for all requests.

**Comment by eile** [comment]
+1 otherwise.

**Pull request merged**